mergify_cli/ci/upload.py

```diff
@@ -61,7 +61,7 @@ async def upload(  # noqa: PLR0913, PLR0917
     files: tuple[str, ...],
     test_language: str | None = None,
     test_framework: str | None = None,
-) -> None:
+) -> list[ReadableSpan]:
```

Comment:
This is too hacky, an upload method should not return things, we should refactor code a bit to separate concerns.

Reply:
https://github.com/Mergifyio/mergify-cli/pull/714